src/org/intellij/erlang/ErlangReferenceContributor.java

```diff
@@ -16,43 +16,78 @@
 
 package org.intellij.erlang;
 
+import com.intellij.patterns.PsiElementPattern;
 import com.intellij.psi.*;
+import com.intellij.util.ObjectUtils;
 import com.intellij.util.ProcessingContext;
+import org.intellij.erlang.psi.ErlangArgumentList;
+import org.intellij.erlang.psi.ErlangFunctionCallExpression;
+import org.intellij.erlang.psi.ErlangMaxExpression;
 import org.intellij.erlang.psi.ErlangQAtom;
 import org.intellij.erlang.psi.impl.ErlangPsiImplUtil;
+import org.intellij.erlang.psi.impl.ErlangPsiImplUtil.ErlangFunctionCallParameter;
+import org.intellij.erlang.psi.impl.ErlangPsiImplUtil.ErlangFunctionCallModuleParameter;
+import org.intellij.erlang.psi.impl.ErlangPsiImplUtil.ErlangFunctionCallFunctionParameter;
 import org.jetbrains.annotations.NotNull;
 
 import static com.intellij.patterns.PlatformPatterns.psiElement;
 
 public class ErlangReferenceContributor extends PsiReferenceContributor {
-  private enum Kind {FUNCTION, MODULE, RECORD}
+  @SuppressWarnings("unchecked")
+  PsiElementPattern.Capture<ErlangQAtom> myAtomArgInFunctionCall = psiElement(ErlangQAtom.class)
```

Comment:
Why is it package-private?

Why have a field for it? Although it's insignificant, but you're wasting memory by it.